django/core/checks/compatibility/django_3_2.py

```diff
@@ -0,0 +1,19 @@
+from __future__ import unicode_literals
+
+from django.conf import settings
+
+from .. import Tags, Warning, register
+
+W001 = Warning(
+    "The DEFAULT_MODEL_PK setting will default to 'django.db.models.BigAutoField "
+    "in Django 4.0. Set this setting to 'django.db.models.AutoField' or define "
+    "an explicit primary key to avoid extra migrations.",
```

Comment:
Should this be part of `hint` instead? Also setting an explicit primary key on models won't actually address this warning as it will keep being emitted until `DEFAULT_MODEL_PK` is set.

I wonder if a per-model warning would be more appropriate when `settings.DEFAULT_MODEL_PK` is not defined , no explicit `primary_key` is specified, and the app config of the model doesn't define `default_model_pk`.

The warning could then hint at either defining the setting, adjusting the app config's `.default_model_pk` or explicitly declaring a `primary_key` on the model warned about.

Reply:
I really like this idea, I've changed the PR to have per-model checks. I'm not 100% on the wording, but there are two cases here: 
1. Old projects with existing models
2. New projects created from templates that don't have a value for `DEFAULT_MODEL_PK`

So I've worded it to say set it to _either_ `BigAutoField` or `AutoField`, in case the previous wording encourages people to just blindly set it to `AutoField` on new projects.